### PR TITLE
Support 16-bit cameras

### PR DESCRIPTION
**Pull request by scpeters**
Adds support for 16-bit cameras, along with a test. Code is from @iche033; I added the test.

### CODE REVIEW

**Comment by scpeters** [comment]
the test doesn't get run by CI, so a reviewer should run it manually with `-DENABLE_DISPLAY_TESTS=True`

File: gazebo_plugins/src/gazebo_ros_camera_utils.cpp [hunk 1]
```diff
@@ -584,9 +594,11 @@ void GazeboRosCameraUtils::PutCameraData(const unsigned char *_src)
     this->image_msg_.header.stamp.sec = this->sensor_update_time_.sec;
     this->image_msg_.header.stamp.nsec = this->sensor_update_time_.nsec;
 
+    unsigned int step = this->skip_* this->width_ *
```
Comment (iche033) [inline gazebo_plugins/src/gazebo_ros_camera_utils.cpp]: I wasn't quite sure what `skip_` represents. I assumed it's the number of channels so put down 1 for `MONO16` and 3 for `RGB16` above. But thinking about it more, it believe it should be the number of bytes per pixel. So it'll be 2 for `MONO16` and 6 for `RGB16`. In this case, `step` would just be `this->skip_*this->width_` (ie. you can keep the old code before the changes here)

Comment (iche033) [reply gazebo_plugins/src/gazebo_ros_camera_utils.cpp]: actually I made these changes. I'll update them now

Comment (iche033) [reply gazebo_plugins/src/gazebo_ros_camera_utils.cpp]: just realized the PR is created from a branch in your repo. Could you apply the following patch? thanks

~~~
diff --git a/gazebo_plugins/src/gazebo_ros_camera_utils.cpp b/gazebo_plugins/src/gazebo_ros_camera_utils.cpp
index d527fb6..c4c7da3 100644
--- a/gazebo_plugins/src/gazebo_ros_camera_utils.cpp
+++ b/gazebo_plugins/src/gazebo_ros_camera_utils.cpp
@@ -414,7 +414,7 @@ void GazeboRosCameraUtils::Init()
   else if (this->format_ == "L16" || this->format_ == "L_INT16")
   {
     this->type_ = sensor_msgs::image_encodings::MONO16;
-    this->skip_ = 1;
+    this->skip_ = 2;
   }
   else if (this->format_ == "R8G8B8" || this->format_ == "RGB_INT8")
   {
@@ -429,7 +429,7 @@ void GazeboRosCameraUtils::Init()
   else if (this->format_ == "R16G16B16" ||  this->format_ == "RGB_INT16")
   {
     this->type_ = sensor_msgs::image_encodings::RGB16;
-    this->skip_ = 3;
+    this->skip_ = 6;
   }
   else if (this->format_ == "BAYER_RGGB8")
   {
@@ -594,11 +594,9 @@ void GazeboRosCameraUtils::PutCameraData(const unsigned char *_src)
     this->image_msg_.header.stamp.sec = this->sensor_update_time_.sec;
     this->image_msg_.header.stamp.nsec = this->sensor_update_time_.nsec;
 
-    unsigned int step = this->skip_* this->width_ *
-        sensor_msgs::image_encodings::bitDepth(this->type_)/8;
     // copy from src to image_msg_
     fillImage(this->image_msg_, this->type_, this->height_, this->width_,
-        step, reinterpret_cast<const void*>(_src));
+        this->skip_*this->width_, reinterpret_cast<const void*>(_src));
~~~

**Review by iche033 — CHANGES_REQUESTED** [review]

**Review by iche033 — APPROVED** [review]

**Comment by scpeters** [comment]
There's an issue with the test: I made a distinct rostest file, but am re-using the camera.cpp file. This allows separate tests to be run, but camera.cpp file now makes two copies of the test results file. I might need to fork camera.cpp into camera16.cpp

**Comment by scpeters** [comment]
I've refactored the camera test in 389c550, with the implementation in camera.h and a distinct test name in camera16bit.cpp

**Comment by j-rivero** [comment]
@osrf-jenkins run tests please

**Comment by j-rivero** [comment]
One of the tests (DistortionTest.cameraDistortionTest) is failing, but is not related to this PR. The merge of https://bitbucket.org/osrf/release-tools/pull-requests/375/enable-display-test-in-gazebo_ros_pkgs triggered this error.

**Comment by scpeters** [comment]
The distortion test fails on my desktop too. I've been trying to figure out the problem

**Comment by scpeters** [comment]
I just rebased on kinetic to get the renamed distortion tests

**Pull request merged**